randovania/gui/widgets/multiplayer_session_users_widget.py

```diff
@@ -363,19 +369,35 @@ def _create_world_item(
                     world_menu.addAction("Connect via debug connector"), self._register_debug_connector, world_id
                 )
 
-        if self.is_admin() or self._session.allow_everyone_claim_world:
-            if owner is None:
-                world_menu.addSeparator()
+        if self.is_admin() or self._session.allow_everyone_claim_world or owner == self.your_id:
+
+            def create_claim_for_each_player_entry() -> None:
+                claim_menu = world_menu.addMenu("Claim for")
+                for p in self._session.users.values():
+                    connect_to(claim_menu.addAction(p.name), self._world_claim_with, world_id, p.id)
+
+            def create_claim_for_yourself_entry() -> None:
                 connect_to(world_menu.addAction("Claim for yourself"), self._world_claim_with, world_id, self.your_id)
 
+            def create_unclaim_entry() -> None:
+                connect_to(world_menu.addAction("Unclaim"), self._world_unclaim, world_id, owner)
+
+            if (
+                owner is None
+                and (self.is_admin() or self._session.allow_everyone_claim_world)
+                and not self._session.users[self.your_id].worlds.get(world_id)
+            ):
+                world_menu.addSeparator()
+                create_claim_for_yourself_entry()
+
                 if self.is_admin():
-                    claim_menu = world_menu.addMenu("Claim for")
-                    for p in self._session.users.values():
-                        connect_to(claim_menu.addAction(p.name), self._world_claim_with, world_id, p.id)
+                    create_claim_for_each_player_entry()
 
-            else:
+            elif owner is not None and (
+                self.is_admin() or self._session.allow_everyone_claim_world or owner == self.your_id
+            ):
                 world_menu.addSeparator()
-                connect_to(world_menu.addAction("Unclaim"), self._world_unclaim, world_id, owner)
+                create_unclaim_entry()
```

Comment:
The Unclaim should show up for your claimed worlds in the "All worlds" section

Reply:
Not so sure about that. If you have a scenario like:
```
Alice (admin)
Bob
|- Prime
Caesar
|- Prime
All Worlds
|- Prime
```
And alice clicks on the actions for prime under the all worlds section, what should happen for her?
If "unclaim" shows, its not quite clear from *who* alice is unclaiming it from. With the world being directly under the user its clear with "ok, i'm unclaiming the world from the user above" which isn't present here.

So IMO either don't make it show at all there (lazy implementation lol), or we show it as "Unclaim from yourself" with an admin option of "unclaim from > [dropdown of every user in the session that has it claimed]" (and maybe an admin option of "unclaim from all users"?)

Reply:
If "Claim for yourself" shows, then "Unclaim for yourself" should also show